server/sonar-server/src/main/java/org/sonar/server/user/ServerUserSession.java

```diff
@@ -52,12 +49,11 @@
   private final DbClient dbClient;
   private final OrganizationFlags organizationFlags;
   private final DefaultOrganizationProvider defaultOrganizationProvider;
-  private final Supplier<List<GroupDto>> groups = Suppliers.memoize(this::loadGroups);
+  private final Supplier<Collection<GroupDto>> groups = Suppliers.memoize(this::loadGroups);
```

Comment:
I think this should be a Set and equals/hashcode should be defined in GroupDto (based on the id)

It's important we know the groups are not ordered, but also as important that there is no duplicate.

WDYT?